src/blueapi/utils/base_model.py

```diff
@@ -17,7 +17,7 @@ class BlueapiModelConfig(BaseConfig):
     allow_population_by_field_name = True
 
 
-class BlueapiPlanModelConfig(BlueapiModelConfig):
+class BlueapiPlanModelConfig(BaseConfig):
```

Comment:
Is this class still needed? Could BlueapiModelConfig be used everywhere instead

Reply:
True if we decide non-plan models shouldn't have camelcase either, as you suggest. I'm on the fence about it.

The reason behind the aliasing in the first place was because we're going to have multiple services in multiple languages so I think there should be _a_ standard among the various APIs. CamelCase seems to be a more common standard for REST etc. from some brief googling I did at the time. It also helps interoperation with Java services and GDA.

We can just say blueapi's API is all snake_case and be done with it, just expect other applications to deal with it, and accept a certain level of divergence. I guess we are already accepting that by making this change...

I'm not sure, any opinions?

Reply:
I'd err on the side of having the API be strict and having the clients adhere to it via whatever mechanisms the de/serialisation libraries offer. If camelCase is more common in rest APIs I wouldn't be as against forcing everything to use camelCase instead, but it'd make the python side more awkward.

[Only tangentially relevant but an interesting read](https://www.ietf.org/archive/id/draft-iab-protocol-maintenance-05.html)

Reply:
The lack of strictness is definitely annoying, and wasn't a problem we had with apischema. I haven't found a way to overcome it with pydantic yet. Maybe it is just worth having an all snake_case API then...
Would also be interested in @joeshannon's thoughts.

Reply:
How did apischema handle it?

Reply:
You can straight up convert rather than alias

Reply:
I think pydantic lets you do that if you set `allow_population_by_field_name` to `False` and then set the alias.

Reply:
I'll take a look at that then, cheers. Although still open to discussion of whether we should be doing any camelCasing at all, may drag us together offline.

Reply:
Hive mind says remove all aliasing

Reply:
Done

Reply:
Back to the original question, do we still need different models for plans vs non-plans?